src/MongoDB.Driver.Core/Core/Authentication/UsernamePasswordCredential.cs

```diff
@@ -40,10 +41,19 @@ public sealed class UsernamePasswordCredential
         public UsernamePasswordCredential(string source, string username, string password)
             : this(source, username, ConvertPasswordToSecureString(password))
         {
+            // Compute saslPreppedPassword immediately and store it securely while the password is already in
+            // managed memory. We don't create a closure over the password so that it will hopefully get 
+            // garbage-collected sooner rather than later.
+            var saslPreppedPassword = ConvertPasswordToSecureString(SaslPrepHelper.SaslPrepStored(password));
+            _saslPreppedPassword = new Lazy<SecureString>(()=>saslPreppedPassword);
```

Comment:
Did your editor not put spaces around "=>"?

If I retyped the ";" Visual Studio reformatted this line as I would have expected.